File: src/modules/role/domain/dto/role_update_dto.py
from ninja import Schema
from typing import List
from src.modules.role.domain.enums.role_references_enum import RoleReferences
from src.shared.domain.enums.status_enum import Status


class RoleUpdateDTO(Schema):
    name: str | None = None
    is_default: bool | None = None
    reference: RoleReferences | None = None
    status: Status | None = None
    abilities_ids: List[int] | None = None

File: src/main/java/br/com/guizzo/projectbasic/modules/role/domain/dto/RoleUpdateDTO.java
package br.com.guizzo.projectbasic.modules.role.domain.dto;

import br.com.guizzo.projectbasic.modules.role.domain.enums.RoleReferences;
import br.com.guizzo.projectbasic.shared.domain.enums.Status;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;

import java.util.List;

@Data
@AllArgsConstructor
public class RoleUpdateDTO {
    private String name;

    private Boolean isDefault;

    private RoleReferences reference;

    private Status status;

    private List<Long> abilitiesIds;
    
}
